### ISSUE TITLE
question regarding the implementation of your temporal contrasting loss

### ISSUE THREAD

**Issue by jaeho3690** [emadeldeen24/TS-TCC#16]
Thank you for your interesting work! I have a question regarding the implementation of your loss implementation on $\mathcal{L}_{TC}^{s}$. I wonder if what I understood is right or please kindly correct me if this is not what you intended.

This is the paper
<img width="565" alt="스크린샷 2023-01-12 오전 9 23 44" src="https://user-images.githubusercontent.com/38973453/211946670-089b3458-a93f-4a60-976d-488c717c08ea.png">

and this is the code
```python
        for i in np.arange(0, self.timestep):
            total = torch.mm(encode_samples[i], torch.transpose(pred[i], 0, 1))
            nce += torch.sum(torch.diag(self.lsoftmax(total)))
        nce /= -1. * batch * self.timestep
```

At first sight, it was hard to understand how the code matches the equation in the paper. To me, it seemed that you have only implemented the numerator part of the equation in the code. However, after some thought, it seemed that the `total` matrix contains elements from both the numerator and denominator. Then by applying a logsoftmax function, you are bounding this matrix to some limit. By only adding the elements in the diagonal terms(=numerator) and adding this as a negative loss, you are essentially making the diagonal terms smaller while making the off-diagnoal terms(=denominator) bigger. This is how I understood. Could you please let me know if this statement is correct? 

thank you!

**Comment by emadeldeen24** [comment]
> the `total` matrix contains elements from both the numerator and denominator.

yes, correct.

The remaining is basically correct too.
